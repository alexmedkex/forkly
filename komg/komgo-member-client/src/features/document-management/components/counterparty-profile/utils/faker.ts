import { CounterpartyProfile, RiskLevel } from '../../../../counterparties/store/types'

const defaultProfile: CounterpartyProfile = {
  id: 'anon-profile-id',
  counterpartyId: 'anon-counterparty-id',
  renewalDate: new Date('2019-07-16').toISOString(),
  riskLevel: RiskLevel.unspecified,
  managedById: 'anon-user-id'
}

export const fakeProfile = (partial?: Partial<CounterpartyProfile>) => {
  return partial ? { ...defaultProfile, ...partial } : defaultProfile
}

export const mockDate = () => {
  const RealDate = Date
  const _GLOBAL: any = global // we love typescript!

  return {
    freeze: (isoDate: string | number | Date) => {
      _GLOBAL.Date = class extends RealDate {
        constructor(...args: any) {
          super()
          return new RealDate(isoDate)
        }
      }
    },
    restore: () => {
      global.Date = RealDate
    }
  }
}
