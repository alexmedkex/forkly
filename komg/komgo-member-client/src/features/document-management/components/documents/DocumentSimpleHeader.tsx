import * as React from 'react'
import { Button } from 'semantic-ui-react'
import styled from 'styled-components'

import { Document } from '../../store/types'
import { FlexHeader } from './DocumentViewHeader'

interface IProps {
  document: Document
  actions: {
    close(): void
  }
}

const StyledFlexHeader = styled(FlexHeader)`
  && {
    justify-content: flex-end;
  }
`

const DocumentSimpleHeader: React.FC<IProps> = (props: IProps) => {
  return (
    <StyledFlexHeader>
      <Button content="Close" onClick={props.actions.close} data-test-id="close-view-document" />
    </StyledFlexHeader>
  )
}

export default DocumentSimpleHeader
