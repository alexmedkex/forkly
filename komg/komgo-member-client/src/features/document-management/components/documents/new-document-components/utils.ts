import { HasId, HasName } from '../../../store/types'

export const hasIdAndNameToOption = (collection: Array<HasId & HasName>): Array<{ key: string }> => {
  const originalCollection = collection.map(el => {
    return { key: el.id, text: el.name, value: el.id }
  })
  return originalCollection.reduce((unique: Array<{ key: string }>, o: { key: string }) => {
    if (!unique.some((obj: { key: string }) => obj.key === o.key)) {
      unique.push(o)
    }
    return unique
  }, [])
}
