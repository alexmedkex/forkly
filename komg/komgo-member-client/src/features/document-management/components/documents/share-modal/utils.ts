import { ICompanyTableItemWithSharedDocs, ICounterpartyWithDocuments } from './SelectedCounterparties'
import { FlagNameValues } from 'semantic-ui-react'
import * as i18nIsoCountries from 'i18n-iso-countries'

export function pluralize(word: string, count: number, singleSuffix: string, pluralSuffix: string) {
  if (count > 1) {
    return `${word}${pluralSuffix}`
  }

  return `${word}${singleSuffix}`
}

export const buildCounterpartyWithDocumentsPickerItems = (
  counterpartiesWithDocs: ICounterpartyWithDocuments[]
): ICompanyTableItemWithSharedDocs[] => {
  i18nIsoCountries.registerLocale(require('i18n-iso-countries/langs/en.json'))
  return counterpartiesWithDocs.map(item => ({
    name: item.counterparty.x500Name.CN,
    countryName: i18nIsoCountries.getName(item.counterparty.x500Name.C.trim(), 'en'),
    country: item.counterparty.x500Name.C.toLocaleLowerCase().trim() as FlagNameValues,
    location: item.counterparty.x500Name.L,
    id: item.counterparty.staticId,
    sharedDocuments: item.documents
  }))
}
