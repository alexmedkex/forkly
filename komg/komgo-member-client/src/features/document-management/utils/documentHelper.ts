import { Document } from '../store/types'

export const isSharedDocument = (doc: Document): boolean => {
  return doc.sharedBy !== 'none'
}

/**
 * It checks if a document have been already reviewed or it returns
 * false if it is pending to be reviewed yet.
 */
export const isReviewed = (doc: Document): boolean => {
  if (isSharedDocument(doc)) {
    return doc.sharedInfo && doc.sharedInfo.status && doc.sharedInfo.status !== 'pending'
  }
}

export const isReviewCompleted = (doc: Document): boolean => {
  if (doc.sharedInfo) {
    return doc.sharedInfo.completedReview
  }
}
