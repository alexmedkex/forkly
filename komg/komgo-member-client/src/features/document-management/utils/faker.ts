import { Document } from '../store/types'
import { mockProduct } from '../store/products/mock-data'
import { mockCategories } from '../store/categories/mock-data'
import { mockDocumentTypes } from '../store/document-types/mock-data'
import { BottomSheetStatus } from '../../bottom-sheet/store/types'
import { DropdownOption } from '../components/documents/my-documents/DocumentListDropdownOptions'

export const MOCK_COMPANY_ID = 'company-id'

export const anonDocument: Document = {
  id: '-1',
  documentId: 'anonDocumentId',
  name: 'anon document',
  product: mockProduct,
  category: mockCategories[2],
  type: mockDocumentTypes[2],
  owner: {
    firstName: 'Owner',
    lastName: 'Owner',
    companyId: MOCK_COMPANY_ID
  },
  hash: 'hash',
  receivedDate: new Date('2000-12-31'),
  registrationDate: new Date('2000-12-31'),
  metadata: [],
  content: undefined,
  sharedWith: [],
  sharedBy: '',
  state: BottomSheetStatus.PENDING
}

export const fakeDocument = (partialDocument?: Partial<Document>): Document => {
  const partial = partialDocument || {}
  return { ...anonDocument, ...partial }
}

const defaultDropdownOption: DropdownOption = {
  key: 'anon',
  text: '',
  value: 'anon',
  content: 'Anon',
  onClick: () => void 0
}

export const fakeDropdownOption = (partial: Partial<DropdownOption>): DropdownOption => {
  return { ...defaultDropdownOption, ...partial }
}
