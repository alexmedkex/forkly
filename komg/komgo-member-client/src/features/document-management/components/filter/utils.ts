import { Category, DocumentType } from '../../index'
import { sortCategories, sortDocumentTypes } from '../../utils/sortingHelper'
import { categoryToColor } from '../documents/document-library/categoryToColor'

export const getDocumentTypeFilterOptions = (categories: Category[], types: DocumentType[]) => {
  return sortCategories(categories).map(c => ({
    name: c.name,
    value: c.id,
    display: {
      color: categoryToColor[c.id]
    },
    items: sortDocumentTypes(types)
      .filter(t => t.category.id === c.id)
      .map(t => ({
        name: t.name,
        value: t.id
      }))
  }))
}
