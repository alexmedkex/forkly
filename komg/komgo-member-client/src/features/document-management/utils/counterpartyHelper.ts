import { Counterparty } from '../../counterparties/store/types'
import { findCounterpartyByStatic } from '../../letter-of-credit-legacy/utils/selectors'

export const counterpartyName = (counterparty: Counterparty): string => {
  return counterparty.x500Name.O
}

export const fetchCounterpartyName = (counterparties: Counterparty[], companyId: string): string => {
  const counterParty = findCounterpartyByStatic(counterparties, companyId)
  return counterParty ? counterParty.x500Name.CN : ''
}
