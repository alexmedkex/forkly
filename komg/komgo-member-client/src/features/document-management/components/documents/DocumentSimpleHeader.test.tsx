import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react'

import DocumentSimpleHeader from './DocumentSimpleHeader'
import { fakeDocument } from '../../utils/faker'

describe('DocumentSimpleHeader', () => {
  const defaultProps = {
    document: fakeDocument(),
    actions: {
      close: jest.fn()
    }
  }

  it('should match default snapshot', () => {
    expect(renderer.create(<DocumentSimpleHeader {...defaultProps} />).toJSON()).toMatchSnapshot()
  })

  it('should call close function when button is clicked', () => {
    const { getByText } = render(<DocumentSimpleHeader {...defaultProps} />)

    fireEvent.click(getByText('Close'))

    expect(defaultProps.actions.close).toHaveBeenCalled()
  })
})
