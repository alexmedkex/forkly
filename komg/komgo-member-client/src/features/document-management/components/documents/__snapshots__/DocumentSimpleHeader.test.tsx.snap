// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentSimpleHeader should match default snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 17px 30px;
}

.c0.c0 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
  className="c0 c1"
>
  <button
    className="ui button"
    data-test-id="close-view-document"
    onClick={[Function]}
    role="button"
  >
    Close
  </button>
</div>
`;
